staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -241,20 +241,28 @@ func (s *APIAggregator) AddAPIService(apiService *apiregistration.APIService, de
 		lister:          s.lister,
 		serviceLister:   s.serviceLister,
 		endpointsLister: s.endpointsLister,
-		delegate:        s.GenericAPIServer.FallThroughHandler,
+		delegate:        s.delegateHandler,
 	}
 	// aggregation is protected
-	s.GenericAPIServer.HandlerContainer.ServeMux.Handle(groupPath, groupDiscoveryHandler)
-	s.GenericAPIServer.HandlerContainer.ServeMux.Handle(groupPath+"/", groupDiscoveryHandler)
+	s.GenericAPIServer.FallThroughHandler.Handle(groupPath, groupDiscoveryHandler)
+	s.GenericAPIServer.FallThroughHandler.UnlistedHandle(groupPath+"/", groupDiscoveryHandler)
 	s.handledGroups.Insert(apiService.Spec.Group)
 }
 
-// RemoveAPIService removes the APIService from being handled.  Later on it will disable the proxy endpoint.
+// RemoveAPIService removes the APIService from being handled.
```

Comment:
doc that this method is not thread-safe, call in the same thread as AddAPIService